.gitlab-ci.yml

```diff
@@ -88,41 +82,30 @@ buildtest:search-with-index-cache:
   stage: buildtest
   image: registry.gitlab.com/magda-data/magda/data61/magda-builder-scala:$CI_COMMIT_REF_SLUG
   retry: 2
-  dependencies:
-    - builders-and-yarn
-    - sbt-prebuild
+  cache:
+    key: SCALA-$SCALA_CACHE_VERSION
+    paths:
+      - "$CI_PROJECT_DIR/sbt-cache"
+      - "$CI_PROJECT_DIR/target"
+      - "$CI_PROJECT_DIR/project/target"
+      - "$CI_PROJECT_DIR/project/project/target"
+      - "$CI_PROJECT_DIR/*/target"
+      - "$CI_PROJECT_DIR/*/project/target"
+      - "$CI_PROJECT_DIR/*/project/project/target"
+      - "$CI_PROJECT_DIR/pip-cache"
   before_script:
-    - |
-      if [ -z "$DOCKER_HOST" -a "$KUBERNETES_PORT" ]; then
-        export DOCKER_HOST='tcp://localhost:2375'
-      fi
     - docker info
     - apk update
     - apk upgrade
     - apk add python3 python3-dev py3-pip build-base openssl-dev libffi-dev
     - pip3 install --upgrade pip
     - pip3 install docker-compose
     - docker-compose --version
-  cache:
-    key: $CI_JOB_NAME-$CACHE_VERSION
-    paths:
-      - "$CI_PROJECT_DIR/pip-cache"
```

Comment:
Any reason we want to remove the pip-cache?
We still can share cache across pipeline only for this step/job (and this cache hardly changes)